spec/classes/init_spec.rb

```diff
@@ -260,6 +260,40 @@
 
   end
 
+  context 'with selboolean specified as a hash' do
+    let(:facts) { { :osfamily => 'RedHat' } }
+    let :params do
+      {
+        :seboolean_hiera_merge => 'false',
+        :seboolean => {
+          'nfs_export_all_ro' => {
+            'value' => 'on',
+          },
+          'nfs_export_all_rw' => {
+            'persistent' => true,
+            'value'      => 'on',
+          },
+        }
+      }
+    end
+
+    it { should contain_class('types') }
+
+    it {
+      should contain_seboolean('nfs_export_all_ro').with({
+        'value' => 'on',
+      })
+    }
+
+    it {
+      should contain_package('nfs_export_all_rw').with({
+        'persistent' => 'true',
+        'value' => 'on',
+      })
+    }
+
+  end
+
```

Comment:
we also need a check if it is specified as an invalid type. Check out https://github.com/ghoneycutt/puppet-module-types/blob/master/spec/classes/init_spec.rb#L273-281